src/integ_test/java/games/strategy/engine/lobby/server/login/LobbyLoginValidatorTest.java

```diff
@@ -117,4 +143,104 @@ public void testLogin() {
     assertNotNull(new LobbyLoginValidator().verifyConnection(challengeProperties, properties,
         Util.createUniqueTimeStamp(), mac, address));
   }
+
+  @Test
+  public void testLogin() {
+    final LobbyLoginValidator validator = new LobbyLoginValidator();
+    final SocketAddress address = new InetSocketAddress(5000);
+    final String name = Util.createUniqueTimeStamp();
+    final String mac = MacFinder.getHashedMacAddress();
+    final String email = "none@none.none";
+    final String password = "foo";
+    new DbUserController().createUser(
+        new DBUser(
+            new DBUser.UserName(name),
+            new DBUser.UserEmail(email)),
+        new HashedPassword(BCrypt.hashpw(Util.sha512(password), BCrypt.gensalt())));
+    final Map<String, String> properties = new HashMap<>();
+    final Map<String, String> challengeProperties = validator.getChallengeProperties(name, address);
+    RsaAuthenticator.appendEncryptedPassword(properties, challengeProperties, password);
+    properties.put(LobbyLoginValidator.LOBBY_VERSION, LobbyServer.LOBBY_VERSION.toString());
+    assertNull(new LobbyLoginValidator().verifyConnection(challengeProperties, properties, name, mac, address));
+    // with a bad password
+    RsaAuthenticator.appendEncryptedPassword(properties, challengeProperties, "wrong");
+    assertNotNull(new LobbyLoginValidator().verifyConnection(challengeProperties, properties, name, mac, address));
+    // with a non existent user
+    assertNotNull(new LobbyLoginValidator().verifyConnection(challengeProperties, properties,
+        Util.createUniqueTimeStamp(), mac, address));
+  }
+
+  @Test
+  public void testLegacyLoginCombined() {
```

Comment:
There looks to be a good bit of redundancy between test cases. The problem I see is that it is not quite obvious what these tests are testing exactly. Even the differences between the test cases can be a bit hard to spot. I recommend looking into some of the tests @ssoloff offered for a nice structure that avoids this problem, eg:
```
public void testLegacyLoginCombined() {
   final Map<String, String> challengeProperties = givenLegacyChallengeProperties();

    RsaAuthenticator.appendEncryptedPassword(properties, challengeProperties, password);
  
      assertNull(new LobbyLoginValidator().verifyConnection(challengeProperties, properties, name, mac, address));
     assertTrue(BCrypt.checkpw(Util.sha512(password), new DbUserController().getPassword(name).value));
}

```

Then, continuing with the example, there would be a second test case to cover these assertions:
```
+    properties.remove(RsaAuthenticator.ENCRYPTED_PASSWORD_KEY);
 +    assertNull(new LobbyLoginValidator().verifyConnection(challengeProperties, properties, name, mac, address));
 +    assertEquals(hashedPassword, new DbUserController().getLegacyPassword(name).value);
```

Which woudl look something like:
```
public void testLegacyLoginCombined() {
   final Map<String, String> challengeProperties = givenBCryptedChallengeProperties();

    RsaAuthenticator.appendEncryptedPassword(properties, challengeProperties, password);
  
 +    assertNull(new LobbyLoginValidator().verifyConnection(challengeProperties, properties, name, mac, address));
 +    assertEquals(hashedPassword, new DbUserController().getLegacyPassword(name).value);
}

```

To paraphrase, summarize, the above structure is pretty straight forward:
```
given
execute test
assertion
```

The 'given' is basically params to the test, and it's interesting that most of the times it can be refactored down to a  one line. Then the 'given' method describes what kind of test data or state you are creating. Params can be added to these methods to make it clear how things are changing , and how our test data is unique, if not for outright labels thanks to the method name. 

Failing all that, please add some comments on what the test strategy is, if this test needs to be updated, it would be a lot of work to figure out how it works.